kolibri/plugins/slideshow_viewer/assets/src/views/SlideshowRendererComponent.vue

```diff
@@ -189,6 +219,26 @@
         */
         this.polyfillSlideObjectFit();
         this.setHooperListWidth();
+        // Do this on nextTick to avoid sliding into position without proper resizing occurring.
+        this.$nextTick(() =>
+          this.$refs.slider.slideTo(this.extraFields.contentState.lastViewedSlideIndex)
+        );
+      },
+      updateContentState() {
+        this.extraFields.contentState.highestViewedSlideIndex = this.highestViewedSlideIndex;
+        this.extraFields.contentState.lastViewedSlideIndex = this.currentSlideIndex;
+        this.$emit('updateContentState', this.extraFields.contentState);
+      },
+      updateProgress() {
```

Comment:
I'm noticing that the the component emits 'updateProgress' whenever you reach the end, even after you've already mastered it. Maybe you could put a guard in here to not do any more progress tracking if the slide has been mastered previously.